R-package/tests/testthat/test_lgb.Booster.R

```diff
@@ -436,10 +436,10 @@ test_that("Saving a model with different feature importance types works", {
         )
     )
 
-    # UNSUPPORTED_IMPORTANCE <- 2L
-    # expect_error({
-    #     model_string <- bst$save_model_to_string(feature_importance_type = UNSUPPORTED_IMPORTANCE)
-    # }, "Unknown importance type")
+    UNSUPPORTED_IMPORTANCE <- 2L
+    expect_error({
+        model_string <- bst$save_model_to_string(feature_importance_type = UNSUPPORTED_IMPORTANCE)
+    }, "Unknown importance type")
```

Comment:
ok I've been able to narrow down the specific valgrind issues tied to this test:

```text
==1314== 
==1314== HEAP SUMMARY:
==1314==     in use at exit: 263,299,633 bytes in 46,121 blocks
==1314==   total heap usage: 2,428,170 allocs, 2,382,049 frees, 5,656,757,258 bytes allocated
==1314== 
==1314== 16 bytes in 1 blocks are definitely lost in loss record 26 of 2,471
==1314==    at 0x483BE63: operator new(unsigned long) (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==1314==    by 0x1519D5CF: __gnu_cxx::new_allocator<unsigned long>::allocate(unsigned long, void const*) (new_allocator.h:115)
==1314==    by 0x1519C0D7: std::allocator_traits<std::allocator<unsigned long> >::allocate(std::allocator<unsigned long>&, unsigned long) (alloc_traits.h:460)
==1314==    by 0x151AEB31: std::_Vector_base<unsigned long, std::allocator<unsigned long> >::_M_allocate(unsigned long) (stl_vector.h:346)
==1314==    by 0x151AE388: std::_Vector_base<unsigned long, std::allocator<unsigned long> >::_M_create_storage(unsigned long) (stl_vector.h:361)
==1314==    by 0x151ACEE8: std::_Vector_base<unsigned long, std::allocator<unsigned long> >::_Vector_base(unsigned long, std::allocator<unsigned long> const&) (stl_vector.h:305)
==1314==    by 0x151AC0A8: std::vector<unsigned long, std::allocator<unsigned long> >::vector(unsigned long, std::allocator<unsigned long> const&) (stl_vector.h:511)
==1314==    by 0x151A4686: LightGBM::GBDT::SaveModelToString[abi:cxx11](int, int, int) const (gbdt_model_text.cpp:349)
==1314==    by 0x15435699: LightGBM::Booster::SaveModelToString[abi:cxx11](int, int, int) const (c_api.cpp:729)
==1314==    by 0x1542923E: LGBM_BoosterSaveModelToString (c_api.cpp:2201)
==1314==    by 0x1544808B: LGBM_BoosterSaveModelToString_R (lightgbm_R.cpp:657)
==1314==    by 0x4941C9C: R_doDotCall (dotcode.c:619)
==1314== 
==1314== 1,008 bytes in 1 blocks are definitely lost in loss record 204 of 2,471
==1314==    at 0x483BE63: operator new(unsigned long) (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==1314==    by 0x15171D59: __gnu_cxx::new_allocator<double>::allocate(unsigned long, void const*) (new_allocator.h:115)
==1314==    by 0x1516E76D: std::allocator_traits<std::allocator<double> >::allocate(std::allocator<double>&, unsigned long) (alloc_traits.h:460)
==1314==    by 0x15169357: std::_Vector_base<double, std::allocator<double> >::_M_allocate(unsigned long) (stl_vector.h:346)
==1314==    by 0x151633D2: std::_Vector_base<double, std::allocator<double> >::_M_create_storage(unsigned long) (stl_vector.h:361)
==1314==    by 0x1515C5BC: std::_Vector_base<double, std::allocator<double> >::_Vector_base(unsigned long, std::allocator<double> const&) (stl_vector.h:305)
==1314==    by 0x1517F65C: std::vector<double, std::allocator<double> >::vector(unsigned long, double const&, std::allocator<double> const&) (stl_vector.h:524)
==1314==    by 0x151A71C2: LightGBM::GBDT::FeatureImportance(int, int) const (gbdt_model_text.cpp:595)
==1314==    by 0x151A49CE: LightGBM::GBDT::SaveModelToString[abi:cxx11](int, int, int) const (gbdt_model_text.cpp:368)
==1314==    by 0x15435699: LightGBM::Booster::SaveModelToString[abi:cxx11](int, int, int) const (c_api.cpp:729)
==1314==    by 0x1542923E: LGBM_BoosterSaveModelToString (c_api.cpp:2201)
==1314==    by 0x1544808B: LGBM_BoosterSaveModelToString_R (lightgbm_R.cpp:657)
==1314== 
==1314== 1,166 (64 direct, 1,102 indirect) bytes in 1 blocks are definitely lost in loss record 219 of 2,471
==1314==    at 0x483BE63: operator new(unsigned long) (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==1314==    by 0x15170A03: __gnu_cxx::new_allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >::allocate(unsigned long, void const*) (new_allocator.h:115)
==1314==    by 0x1516C719: std::allocator_traits<std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >::allocate(std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >&, unsigned long) (alloc_traits.h:460)
==1314==    by 0x15166645: std::_Vector_base<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >::_M_allocate(unsigned long) (stl_vector.h:346)
==1314==    by 0x1516139C: std::_Vector_base<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >::_M_create_storage(unsigned long) (stl_vector.h:361)
==1314==    by 0x1515A650: std::_Vector_base<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >::_Vector_base(unsigned long, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > const&) (stl_vector.h:305)
==1314==    by 0x151567C4: std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >::vector(unsigned long, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > const&) (stl_vector.h:511)
==1314==    by 0x151A4640: LightGBM::GBDT::SaveModelToString[abi:cxx11](int, int, int) const (gbdt_model_text.cpp:348)
==1314==    by 0x15435699: LightGBM::Booster::SaveModelToString[abi:cxx11](int, int, int) const (c_api.cpp:729)
==1314==    by 0x1542923E: LGBM_BoosterSaveModelToString (c_api.cpp:2201)
==1314==    by 0x1544808B: LGBM_BoosterSaveModelToString_R (lightgbm_R.cpp:657)
==1314==    by 0x4941C9C: R_doDotCall (dotcode.c:619)
==1314== 
==1314== 8,193 bytes in 1 blocks are definitely lost in loss record 1,104 of 2,471
==1314==    at 0x483BE63: operator new(unsigned long) (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==1314==    by 0x73513AF: std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::reserve(unsigned long) (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.28)
==1314==    by 0x7346E4B: std::__cxx11::basic_stringbuf<char, std::char_traits<char>, std::allocator<char> >::overflow(int) (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.28)
==1314==    by 0x734F679: std::basic_streambuf<char, std::char_traits<char> >::xsputn(char const*, long) (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.28)
==1314==    by 0x73419B3: std::basic_ostream<char, std::char_traits<char> >& std::__ostream_insert<char, std::char_traits<char> >(std::basic_ostream<char, std::char_traits<char> >&, char const*, long) (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.28)
==1314==    by 0x151A4946: LightGBM::GBDT::SaveModelToString[abi:cxx11](int, int, int) const (gbdt_model_text.cpp:363)
==1314==    by 0x15435699: LightGBM::Booster::SaveModelToString[abi:cxx11](int, int, int) const (c_api.cpp:729)
==1314==    by 0x1542923E: LGBM_BoosterSaveModelToString (c_api.cpp:2201)
==1314==    by 0x1544808B: LGBM_BoosterSaveModelToString_R (lightgbm_R.cpp:657)
==1314==    by 0x4941C9C: R_doDotCall (dotcode.c:619)
==1314==    by 0x494CCC6: do_dotcall (dotcode.c:1281)
==1314==    by 0x499FB01: bcEval (eval.c:7078)
==1314== 
==1314== 1,048,576 bytes in 1 blocks are definitely lost in loss record 2,429 of 2,471
==1314==    at 0x483BE63: operator new(unsigned long) (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==1314==    by 0x15166A37: __gnu_cxx::new_allocator<char>::allocate(unsigned long, void const*) (new_allocator.h:115)
==1314==    by 0x151603C0: std::allocator_traits<std::allocator<char> >::allocate(std::allocator<char>&, unsigned long) (alloc_traits.h:460)
==1314==    by 0x15166783: std::_Vector_base<char, std::allocator<char> >::_M_allocate(unsigned long) (stl_vector.h:346)
==1314==    by 0x15160102: std::_Vector_base<char, std::allocator<char> >::_M_create_storage(unsigned long) (stl_vector.h:361)
==1314==    by 0x151595DC: std::_Vector_base<char, std::allocator<char> >::_Vector_base(unsigned long, std::allocator<char> const&) (stl_vector.h:305)
==1314==    by 0x151548A0: std::vector<char, std::allocator<char> >::vector(unsigned long, std::allocator<char> const&) (stl_vector.h:511)
==1314==    by 0x15448018: LGBM_BoosterSaveModelToString_R (lightgbm_R.cpp:656)
==1314==    by 0x4941C9C: R_doDotCall (dotcode.c:619)
==1314==    by 0x494CCC6: do_dotcall (dotcode.c:1281)
==1314==    by 0x499FB01: bcEval (eval.c:7078)
==1314==    by 0x498B67F: Rf_eval (eval.c:727)
==1314== 
==1314== LEAK SUMMARY:
==1314==    definitely lost: 1,057,857 bytes in 5 blocks
==1314==    indirectly lost: 1,102 bytes in 2 blocks
==1314==      possibly lost: 0 bytes in 0 blocks
==1314==    still reachable: 262,240,674 bytes in 46,114 blocks
==1314==                       of which reachable via heuristic:
==1314==                         newarray           : 4,264 bytes in 1 blocks
==1314==         suppressed: 0 bytes in 0 blocks
==1314== Reachable blocks (those to which a pointer was found) are not shown.
==1314== To see them, rerun with: --leak-check=full --show-leak-kinds=all
==1314== 
==1314== For lists of detected and suppressed errors, rerun with: -s
==1314== ERROR SUMMARY: 7 errors from 6 contexts (suppressed: 0 from 0)
valgrind found 1057857 bytes definitely lost
Error: Process completed with exit code 255.
```

similar to https://github.com/microsoft/LightGBM/pull/3443#discussion_r505617009, I think we should add a `testthat::skip()` on this test, and I'll document it in an issue that we can come back to in the future.

I think it's important to get the other fixes from this PR merged to improve the user experience, and using `testthat::skip()` should avoid the errors on CRAN. I'm also ok with that because I believe the valgrind errors here might just be the result of how `expect_error()` works.